std/haxe/ds/Vector.hx

```diff
@@ -61,7 +61,7 @@ abstract Vector<T>(VectorData<T>) {
 
 		If `length` is less than or equal to 0, the result is unspecified.
 	**/
-	public inline function new(length:Int) {
+	extern overload public inline function new(length:Int) {
```

Comment:
just out of curiosity why is the extern keyword needed? I am actually surprised that you can declare a function body for a function marked as external.

Reply:
AFAIK `extern inline` means that function will be inlined and will not keeped in generated code when you use reflection. Currently overload only allowed in `extern` mode.
Also, `extern inline new` for classes means that `inline cannot be canceled`, so inline constructor cannot be saved as instance in variable, compiler will error about that. Looks like there is no error for abstracts tho, so it should work.

Reply:
> Looks like there is no error for abstracts tho, so it should work.

Might be a good idea to add a test for this somewhere, since this is the whole point of using an overload instead of #9478

Reply:
>  so inline constructor cannot be saved as instance in variable

Ah, sorry I misunderstood exactly what you were talking about. This is what I had in mind:
```haxe
final create = Vector.new;
final vector = create(10);
```
This works with this overload implementation, but if we used the default argument implementation from #9478 it would have been broken (`Not enough arguments, expected value:Null<Int>`).